spec/classes/foreman_proxy__config__spec.rb

```diff
@@ -352,6 +352,20 @@
             it 'should create shim.efi symlink for Red Hat version 6 and older' do
               should contain_file('/var/lib/tftpboot/grub/shim.efi').with_ensure('link')
             end
+            case facts[:operatingsystem]
+              when /^(RedHat|Scientific|OracleLinux)$/
+                it 'should copy grub.efi for Red Hat and clones' do
+                  should contain_file('/var/lib/tftpboot/grub/grubx64.efi').with_source('/boot/efi/EFI/redhat/grub.efi')
+                end
+              when 'Fedora'
+                it 'should copy grub.efi for Red Hat and clones' do
+                  should contain_file('/var/lib/tftpboot/grub/grubx64.efi').with_source('/boot/efi/EFI/fedora/grub.efi')
+                end
+              when 'CentOS'
+                it 'should copy grub.efi for Red Hat and clones' do
+                  should contain_file('/var/lib/tftpboot/grub/grubx64.efi').with_source('/boot/efi/EFI/centos/grub.efi')
```

Comment:
This should be redhat since this is EL6. That means you can merge it into the RH block.